Menlo-Legacy/lib/Menlo/CLI/Compat.pm

```diff
@@ -1985,10 +1985,19 @@ sub build_stuff {
     push @{$dist->{want_phases}}, 'develop' if $self->{with_develop} && $depth == 0;
     push @{$dist->{want_phases}}, 'configure' if $self->{with_configure} && $depth == 0;
 
-    my @deps = $self->find_prereqs($dist);
     my $module_name = $self->find_module_name($configure_state) || $dist->{meta}{name};
     $module_name =~ s/-/::/g;
 
+    my @all_deps = $self->find_prereqs($dist);
+    my @deps = grep { $_->module ne $module_name } @all_deps;
```

Comment:
Will depending on `SelfDep::SubPackage` from the same dist trick this check?

Reply:
Indeed it would; well-spotted.  I was concerned about this as well, but thought it better to catch at least some cases rather than none.

I've pushed some changes to address the sub-package case: now, the code uses `Module::Metadata->package_versions_from_directory` to enumerate all packages in `$dist->{dir}`.  This should catch the `SelfDep::SubPackage` case, but I'm worried that the logic is now overbroad and catches things like modules that aren't exposed for public consumption (e.g. test-suite-only stuff).

Reply:
> I'm worried that the logic is now overbroad and catches things like modules that aren't exposed for public consumption (e.g. test-suite-only stuff).

On reflection, this probably isn't a big deal.

Reply:
Yep. You can hide from the metacpan indexer with:
```perl
package # hide from indexer
  MyApp::Secret;
```

I assume the same trick works elsewhere